### PR TITLE
[IMP] hw_drivers: replace urllib3 by requests

### PR DESCRIPTION
**Pull request by loouis-t**
In order to make it standard between the IoT Box code and the rest of Odoo, we replaced network fetches using `urllib3` by `requests` calls.

Task: 3619602

### CODE REVIEW

**Comment by loouis-t** [comment]
@odoo/rd-security could you have a look at this please?

## Context

We still used `urllib3` to fecth from the IoT Box.

### Solution proposed

We replaced those calls using the `requests` library.

## Security issues

- `response = requests.post(`
  - **Reason for the use:** Need to fetch the db 
  - **Safe because:** all `request.post/get` use hardcoded urls
- `path = path_file(str(Path().joinpath(*drivers_path)))`
  - **Safe because:** all the lines triggering security review, that are not about `requests` are just lines where indentation changed.

File: addons/hw_drivers/tools/helpers.py [hunk 1]
```diff
@@ -20,7 +19,7 @@
 import socket
 import subprocess
 from urllib.parse import parse_qs
-import urllib3
+import urllib3.util
```
Comment (lse-odoo) [inline addons/hw_drivers/tools/helpers.py:22]: what is the point of this import ?

**Review by Ysoroko — CHANGES_REQUESTED** [review]
What about the other code in IoT using urllib3?
`connection_manager.py`
`server_logger.py`
`DisplayDriver.py`
etc.

**Review by xmo-odoo — COMMENTED** [review]
@robodoo override=ci/security

File: addons/hw_drivers/tools/helpers.py (hunk 1 above)
Comment (loouis-t) [reply addons/hw_drivers/tools/helpers.py:22]: we need it to use the method: `urllib3.util.parse_url()` on line 642.

**Review by Ysoroko — APPROVED** [review]
LGTM for all the:
 - Display driver
 - Odoo version lookup
 - Pairing Code lookup
 - Keyboard layouts
 - IoT Handlers download

Tested and works as expected

**Comment by Ysoroko** [comment]
@robodoo r+

**Pull request closed without merging**